### PR TITLE
File

### PR DESCRIPTION
**Pull request by tal2301**

### CODE REVIEW

File: src/file/storage/storage.service.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { fileModel } from '../file.model';
+
+export class storageService {
+
+  public static delete(filePath: fs.PathLike) {
+    return fs.unlink(filePath, (err) => { if (err) throw err; });
+  }
+
+  public static update(filePath: String) {
+    // TODO: Change filePath in S3 (fileName) (Necessary?)
```
Comment (Shahar-Y) [inline src/file/storage/storage.service.ts]: Open an issue

File: src/share/share.validator..ts [hunk 2]
```diff
@@ -1,13 +0,0 @@
-import Validator from '../validator';
```
Comment (Shahar-Y) [inline src/share/share.validator..ts]: Y U do dis?

File: src/file/file.controller.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    return fileService.findByCreationDate(fromDate, toDate);
+  }
+
+  public static async delete(fileId: string): Promise<IFile> {
+    const currFile: IFile = await fileController.findById(fileId);
+    const ret = await fileService.delete(fileId);
+    if (ret) {
+      storageService.delete(currFile.path);
+    }
+    // TODO: Add if not deleted from storage
```
[the remark this answers is not in the document]
Comment (tal2301) [reply src/file/file.controller.ts]: Issue 48

File: src/file/file.router.spec.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    const removeCollectionPromises = [];
+    for (const i in mongoose.connection.collections) {
+      removeCollectionPromises.push(mongoose.connection.collections[i].remove({}));
+    }
+    await Promise.all(removeCollectionPromises);
+    // await fileModel.remove({}, (err: Error) => {});
+  });
+
+  beforeEach(async () => {
+    // Remove files from DB
```
[the remark this answers is not in the document]
Comment (tal2301) [reply src/file/file.router.spec.ts]: Deal with it!! :D

File: src/file/storage/storage.service.ts (hunk 1 above)
Comment (tal2301) [reply src/file/storage/storage.service.ts]: Issue 49

File: src/share/share.validator..ts (hunk 2 above)
Comment (tal2301) [reply src/share/share.validator..ts]: Yonatan told me to O_O

File: src/file/file.controller.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    return ret;
+  }
+
+  public static async update(fileId: string, file: Partial<IFile>): Promise<IFile> {
+    const currFile = await fileService.update(fileId, file);
+    if (currFile) {
+      storageService.update(currFile.path);
+    }
+    // TODO: Add if not updated in storage
```
[the remark this answers is not in the document]
Comment (tal2301) [reply src/file/file.controller.ts]: Issue 48

File: src/file/file.router.spec.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Shahar-Y) [reply src/file/file.router.spec.ts]: FIRED

File: src/share/share.validator..ts (hunk 2 above)
Comment (Shahar-Y) [reply src/share/share.validator..ts]: OK

File: src/file/file.responder.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    if (ret) {
+      return res.json({ success: true, return: ret });
+    }
+    throw new FileErrors.NoFilesFoundError();
+  }
+
+  static async delete(req: express.Request, res: express.Response) {
+    const ret = await fileController.delete(req.params.id);
+    if (ret) {
+      return res.json({ success: true, return: ret });
```
[the remark this answers is not in the document]
Comment (tal2301) [reply src/file/file.responder.ts]: ![alt-text]https://media.giphy.com/media/8vFwTyU8iwZilmMFIQ/giphy.gif

[the remark this answers is not in the document]
Comment (tal2301) [reply src/file/file.responder.ts]: alt-text

[the remark this answers is not in the document]
Comment (tal2301) [reply src/file/file.responder.ts]: alt text

[the remark this answers is not in the document]
Comment (tal2301) [reply src/file/file.responder.ts]: alt text

[the remark this answers is not in the document]
Comment (tal2301) [reply src/file/file.responder.ts]: ![]https://media.giphy.com/media/8vFwTyU8iwZilmMFIQ/giphy.gif

[the remark this answers is not in the document]
Comment (tal2301) [reply src/file/file.responder.ts]: Alt Text

[the remark this answers is not in the document]
Comment (tal2301) [reply src/file/file.responder.ts]: <iframe src="https://giphy.com/embed/8vFwTyU8iwZilmMFIQ" width="480" height="480" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/cbc-funny-comedy-8vFwTyU8iwZilmMFIQ">via GIPHY</a></p>

[the remark this answers is not in the document]
Comment (tal2301) [reply src/file/file.responder.ts]: OK

[the remark this answers is not in the document]
Comment (Shahar-Y) [reply src/file/file.responder.ts]: 🤣

File: uploads/config.ts-1533136308728 [hunk 7]
```diff
@@ -0,0 +1,63 @@
+/**
```
Comment (Shahar-Y) [inline uploads/config.ts-1533136308728]: What is this file for? 
And what's with the name?

**Pull request merged**